--- a/components/helium_services/extension_ids.h
+++ b/components/helium_services/extension_ids.h
@@ -20,6 +20,13 @@
     // SHA1(kUBlockOriginComponentId)
     "70C2895ED505C532A80761F0928B821B103D0786";
 
+inline constexpr char kTwpComponentId[] =
+    "ocjhnlhmeeocanachmegpeebocpolclb";
+
+inline constexpr char kTwpComponentIdHashed[] =
+    // SHA1(kTwpComponentId)
+    "66FB7FBFA92C835B1D94324A55C8F466BE48933F";
+
 }  // namespace helium
 
 #endif  /* COMPONENTS_HELIUM_SERVICES_EXTENSION_IDS_H_ */
--- a/chrome/browser/extensions/component_loader.h
+++ b/chrome/browser/extensions/component_loader.h
@@ -219,6 +219,7 @@
                                  const std::string& description_string);
   void AddWebStoreApp();
   void AddUBlock();
+  void AddTwp();
 
 #if BUILDFLAG(IS_CHROMEOS)
   void AddChromeApp();
--- a/chrome/browser/extensions/component_loader.cc
+++ b/chrome/browser/extensions/component_loader.cc
@@ -43,5 +43,6 @@
 #include "ui/base/resource/resource_bundle.h"
 #include "third_party/ublock/resources/grit/ublock_resources.h"
+#include "third_party/twp/resources/grit/twp_resources.h"
 
 #if BUILDFLAG(IS_CHROMEOS)
 #include "ash/constants/ash_features.h"
@@ -455,6 +456,21 @@
   }
 }
 
+void ComponentLoader::AddTwp() {
+  Add(IDR_TWP_MANIFEST_JSON, base::FilePath(FILE_PATH_LITERAL("twp")));
+
+  using namespace extension_l10n_util;
+  ComponentExtensionInfo& info = component_extensions_.back();
+  std::string error;
+  bool localized = LocalizeExtension(info.root_directory, &info.manifest,
+                      GzippedMessagesPermission::kAllowForTrustedSource,
+                      &error, true);
+
+  if (!localized) {
+    LOG(ERROR) << "Failed to localize TWP: " << error;
+  }
+}
+
 #if BUILDFLAG(IS_CHROMEOS)
 void ComponentLoader::AddChromeApp() {
   AddWithNameAndDescription(
@@ -533,6 +549,7 @@
   if (!skip_session_components) {
     AddWebStoreApp();
     AddUBlock();
+    AddTwp();
 #if BUILDFLAG(IS_CHROMEOS)
     AddChromeApp();
 #endif  // BUILDFLAG(IS_CHROMEOS)
--- a/chrome/browser/extensions/component_extensions_allowlist/allowlist.cc
+++ b/chrome/browser/extensions/component_extensions_allowlist/allowlist.cc
@@ -21,1 +21,2 @@
 #include "third_party/ublock/resources/grit/ublock_resources.h"
+#include "third_party/twp/resources/grit/twp_resources.h"
@@ -38,2 +39,3 @@
       helium::kUBlockOriginComponentId,
+      helium::kTwpComponentId,
 #if BUILDFLAG(IS_CHROMEOS)
@@ -99,2 +101,3 @@
     case IDR_UBLOCK_MANIFEST_JSON:
+    case IDR_TWP_MANIFEST_JSON:
 #if BUILDFLAG(IS_CHROMEOS)
